src/components/visualizations/chart/colorFactory.ts

```diff
@@ -66,24 +69,30 @@ export abstract class ColorStrategy implements IColorStrategy {
         stackByAttribute: any,
         afm: AFM.IAfm
     ) {
-        this.colorAssignment = this.createColorMapping(
+        const { fullColorAssignment, outputColorAssignment } = this.createColorAssignment(
             colorPalette,
             colorMapping,
             measureGroup,
             viewByAttribute,
             stackByAttribute,
             afm
         );
+        this.fullColorAssignment = fullColorAssignment;
+        this.outputColorAssignment = outputColorAssignment ? outputColorAssignment : fullColorAssignment;
 
-        this.palette = this.createPalette(colorPalette, this.colorAssignment, viewByAttribute, stackByAttribute);
+        this.palette = this.createPalette(colorPalette, this.fullColorAssignment, viewByAttribute, stackByAttribute);
     }
 
     public getColorByIndex(index: number): string {
```

Comment:
Move function getColorFromMapping(mappingHeader: IMappingHeader, colorMapping: IColorMapping[]
    ): IColorItem 

export function getColorByGuid(colorPalette: IColorPalette, guid: string, index: number) 

function getRgbStringFromRGB(color: IRGBColor) {

to colors utils